node/src/accountant/logging_utils/accounting_msgs_debug.rs

```diff
@@ -0,0 +1,467 @@
+        &mut self,
+        msg_type: AccountingMsgType,
+        new_postings: Vec<NewPosting>,
+        log_window_size: u16,
+    ) -> Option<LoggableStats> {
+        self.record(new_postings, msg_type);
+        self.request_log_instruction(log_window_size, msg_type)
+    }
+
+    fn record(&mut self, new_postings: Vec<NewPosting>, msg_type: AccountingMsgType) {
+        match msg_type {
+            AccountingMsgType::RoutingServiceProvided => {
+                self.report_routing_service_provided_processed
+                    .handle_new_postings(new_postings);
+            }
+            AccountingMsgType::ExitServiceProvided => {
+                self.report_exit_service_provided_processed
+                    .handle_new_postings(new_postings);
+            }
+            AccountingMsgType::ServicesConsumed => {
+                self.report_services_consumed_processed
+                    .handle_new_postings(new_postings);
+            }
+        }
+    }
+
+    fn request_log_instruction(
+        &mut self,
+        gap_size: u16,
+        msg_type: AccountingMsgType,
+    ) -> Option<LoggableStats> {
+        match msg_type {
+            AccountingMsgType::RoutingServiceProvided => self
+                .report_routing_service_provided_processed
+                .loggable_stats(gap_size),
+            AccountingMsgType::ExitServiceProvided => self
+                .report_exit_service_provided_processed
+                .loggable_stats(gap_size),
+            AccountingMsgType::ServicesConsumed => self
+                .report_services_consumed_processed
+                .loggable_stats(gap_size),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub struct NewPosting {
+    address: Address,
+    amount_wei: u128,
+}
+
+impl NewPosting {
+    pub fn new(address: Address, amount_wei: u128) -> Self {
+        Self {
+            address,
+            amount_wei,
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct LoggableStats {
+    msg_type: AccountingMsgType,
+    accounting_msg_stats: HashMap<Address, u128>,
+    log_window_in_pcs_of_msgs: u16,
+}
+
+impl Display for LoggableStats {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let label = format!(
+            "Account debits in last {} {:?} messages (wei):",
+            self.log_window_in_pcs_of_msgs, self.msg_type
+        );
+        let stats = self
+            .accounting_msg_stats
+            .iter()
+            .sorted()
+            .map(|(address, sum)| format!("{:?}: {}", address, sum))
+            .collect_vec();
+        once(label).chain(stats).join("\n").fmt(f)
+    }
+}
+
+#[derive(Default)]
+struct AccountingMsgStats {
+    stats: HashMap<Address, u128>,
+    msg_count_since_last_logged: usize,
+}
+
+impl AccountingMsgStats {
+    fn loggable_stats(&mut self, log_window_size: u16) -> Option<LoggableStats> {
+        if self.msg_count_since_last_logged == log_window_size as usize {
```

Comment:
Let's move this to a different fn, and the fn should be called `should_log_stats()`. Like this:

```rust
fn should_log_stats(&self, log_window_size: u16) -> bool {
    self.msg_count_since_last_logged >= log_window_size as usize
}
```